Rewrite vc in Python

winpat opened the pull request.

winpat committed:
Remove Go code

winpat committed:
Bump vault version

winpat committed:
Add MVP written in Python

winpat committed:
Fix recursive option on list command

Without the negation argument it does not behave like a boolean flag.

winpat committed:
Allow to load .vaultrc from different locations

winpat committed:
Add function to update token in config file

winpat committed:
Use tls settings from config file

winpat committed:
Fallback to kv v1 on old engines which do not use options

winpat committed:
Add login command

winpat committed:
Rename src directory to vc

winpat committed:
Add pytest with a few useful fixtures

winpat committed:
Add parameterized test for the show command

winpat committed:
Verify that commands deal correctly with invalid mountpoints

winpat committed:
Rename test

winpat committed:
Refactor if statement

winpat committed:
Reformat code with black

winpat committed:
Refactor logic to split k/v pairs

winpat committed:
Add additional tests

winpat committed:
Add more command tests

winpat committed:
Add pytest dependencies to requirements.txt

winpat committed:
Update sample configuration

winpat committed:
Configure travis for python

karras commented:
Might trigger a PR to update the Vault docs: https://www.vaultproject.io/api/libraries.html

:)

winpat committed:
Add missing import

winpat committed:
Add dialog to create config if does not exist

winpat committed:
Add future-fstrings package

winpat committed:
Add minimal setup.py

winpat committed:
Downgrade pytest for python 3.4 support

winpat committed:
Hardwire vc config in pytest fixture

winpat committed:
Catch exception if token is missing

winpat committed:
Remove debug statement

winpat committed:
Adjust travis badges to private fork

winpat committed:
Update travis config

winpat committed:
Add Python 3.4 compatibility

winpat committed:
Add entrypoint to setup.py

winpat committed:
Add docstrings to cli commands

winpat committed:
Improve error message if config file does not contain a token

winpat committed:
Add support for userpass authentication backend (WIP)

winpat committed:
Don't set auth token if command is "login"

winpat committed:
Set list command default value for path

winpat committed:
Handle list on empty backends gracefully

winpat committed:
Test userpass login

winpat committed:
Add pytest-isort and resort imports

winpat committed:
Add pytest-black and reformat code

winpat committed:
Add pytest-flake8 and fix linting issues

winpat committed:
Add pytest-black to requirements.txt

winpat committed:
Catch LoginFailed exception

winpat committed:
Add option to ls command to display absolute path

This is required to make the bash autocompletion a bit easier.

winpat committed:
Improve dynamic candidates for bash autcompletion

winpat committed:
Run flake8 in CI

winpat committed:
Change url of Travis CI shields/badges

winpat committed:
Update example config file

winpat committed:
Update README.md

winpat committed:
Add unreleased section to CHANGELOG

winpat committed:
Fix typo

winpat committed:
Turn _get_kv_backends into a lazy property

This will reduce the amount of API calls through commands like `ls -r` quite a bit.
